peachjam/urls.py

```diff
@@ -206,8 +206,8 @@
         ArticleTopicListView.as_view(),
         name="article_topic_list",
     ),
-    path(
-        "articles/<str:date>/<str:author>/<slug:slug>",
+    re_path(
+        "^articles/(?P<date>[0-9]{4}-[0-9]{2}-[0-9]{2})/(?P<author>[-a-zA-Z0-9_]+)/(?P<slug>[-a-zA-Z0-9_]+)$",
```

Comment:
It think this allows `/articles/9999-88-77/...` which still isn't valid. I don't think you can escape parsing the date in the view, catching an exception on an invalid value, and returning a 404.

Put another way: you always need to validate date URL params in some way :)

Reply:
@longhotsummer I went with a custom path converter as shown here https://docs.djangoproject.com/en/4.2/topics/http/urls/#registering-custom-path-converters
I think its a better place to do the url validation rather than in the view itself